### PR TITLE
ambientlight: Add ambient light API simulator

### PR DESCRIPTION
**Pull request by MatonAnthony**
Add an Ambient Light API simulator. (see #5)

Signed-off-by: Anthony Maton <me@anthony-maton.me>

### CODE REVIEW

**Comment by rzr** [comment]
Thx @MatonAnthony ... looks good just a remark regarding the value

maybe there is no semantic needed at the driver level...

value could just be a number to align: 
https://github.com/SamsungInternet/color-sensor-js/blob/master/lib/simulator.js#L20

this should be also fixed in geolocation's simulator @Vinnl 

What are you opinions on it ?

**Comment by rzr** [comment]
Next step is to integrate it to lib/ambientlight/index.js and use it by default see:
https://github.com/rzr/color-sensor-js/blob/master/lib/index.js

Feel free to refactor your file if you want to align to the actual driver...

File: lib/ambientlight/index.js [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 var console = require('console');
 var BH1750 = require('bh1750');
```
Comment (rzr) [inline lib/ambientlight/index.js]: thuis could be sourounded by a try to load on your macos

File: lib/ambientlight/index.js [hunk 2]
```diff
@@ -77,6 +78,9 @@ AmbientLight.prototype.start = function start() {
   this.state = 'activating';
   if (!this.sensor) {
     try {
+      if (this.options.controller === undefined) {
+        this.sensor = new Simulator();
+      }
       if (this.options.controller === 'bh1750') {
```
Comment (rzr) [inline lib/ambientlight/index.js]: else ..

**Review by rzr — APPROVED** [review]
looks good could be improved now or later

try to run it
using

```sh
node lib/ambiantlight
```

File: lib/ambientlight/index.js [hunk 3]
```diff
@@ -42,7 +46,7 @@ function AmbientLight(options) {
 
   this.options = options || {};
   this.options.frequency = this.options.frequency || 1;
-  this.options.controller = this.options.controller || 'bh1750';
+  this.options.controller = this.options.controller;
```
Comment (rzr) [inline lib/ambientlight/index.js]: || 'simulator'

File: lib/ambientlight/index.js [hunk 4]
```diff
@@ -56,7 +60,7 @@ AmbientLight.prototype.update = function update() {
         return self.onerror(data);
       } else {
         self.timestamp = new Date();
-        self.illuminance = Number(data);
+        self.illuminance = Number(data.illuminance);
```
Comment (rzr) [inline lib/ambientlight/index.js]: I should remove illuminance from simulator and preserve the original line to keep compatibility with bh driver

File: lib/ambientlight/simulator.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ function Simulator () {
   };
 }
 
-Simulator.prototype.read = function (callback) {
+Simulator.prototype.readLight = function (callback) {
```
Comment (rzr) [inline lib/ambientlight/simulator.js:25]: ok you aligned to sensor good maybe sensor should use read, but upstream is not very reactive

please squash this change in similator.js change

File: package.json [hunk 6]
```diff
@@ -13,15 +13,16 @@
...
   },
+  "optionalDependencies": {
+    "bh1750": "^0.0.5",
+    "bmp085-sensor": "^0.0.5"
+  },
   "keywords": [
     "ambiant",
     "ambiantlight",
     "generic",
-    "i2c",
```
Comment (rzr) [inline package.json]: this keyword can be kept as it can be supported

**Review by rzr — COMMENTED** [review]
Ok it's improving, have you tested it on macos ?

**Comment by MatonAnthony** [comment]
Yes, I tested it on macos.

File: lib/ambientlight/simulator.js (hunk 5 above)
Comment (MatonAnthony) [reply lib/ambientlight/simulator.js:25]: I intend to rebase everything into a single commit once it's ready for merge to have a clear commit easy to revert.

**Review by rzr — COMMENTED** [review]
now It looks good, 

maybe the temp driver should not be optional as simulator is not yet done...

now you can try to squash some commits to hide intermediary works

**Review by rzr — APPROVED** [review]
all good, same should be done for temperature 
then I will release a npm

**Pull request merged**